### ISSUE TITLE
Support http-stream protocol in substrate/native java functions

### ISSUE THREAD

**Issue by zootalures** [fnproject/fdk-java#164]
We are planning to deprecate all formats except for the unix-socket based http-stream format in the coming months 

for standard JVM functions this requires some JNI  to create unix sockets for substrate the functions either need to be linked against this (possibly by compiling the unix socket library against a different target)

**Comment by tzezula** [comment]
I think we can use the JNI library from the substrate.
But we will not be able to do `static` image anymore.

**Comment by zootalures** [comment]
If I compiled the existing JNI code as a static lib is there a way we could make that work?

**Comment by tzezula** [comment]
Yes, this will work but it will require special interfaces for calling to C functions.

Someting like:
```
interface MyCFunction extends CFunctionPointer {
  @InvokeCFunctionPointer
  public int call(...);
}
```
I will need to think a bit about this what's better.

**Comment by rikgibson** [comment]
@tzezula can this be closed now or do you still have spark to do?
